plugins/git/git.plugin.zsh

```diff
@@ -1,177 +1,212 @@
-  if ! [ -z $1 ]; then
-    git log --pretty=$1
-  fi
-}
+alias gl='git pull'
+alias glg='git log --stat --color'
+alias glgp='git log --stat --color -p'
+alias glgg='git log --graph --color'
+alias glgga='git log --graph --decorate --all'
+alias glgm='git log --graph --max-count=10'
+alias glo='git log --oneline --decorate --color'
+alias glol="git log --graph --pretty=format:'%Cred%h%Creset -%C(yellow)%d%Creset %s %Cgreen(%cr) %C(bold blue)<%an>%Creset' --abbrev-commit"
+alias glola="git log --graph --pretty=format:'%Cred%h%Creset -%C(yellow)%d%Creset %s %Cgreen(%cr) %C(bold blue)<%an>%Creset' --abbrev-commit --all"
+alias glog='git log --oneline --decorate --color --graph'
+alias glone='git clone'
 alias glp="_git_log_prettily"
 compdef _git glp=git-log
 
-# Work In Progress (wip)
-# These features allow to pause a branch development and switch to another one (wip)
-# When you want to go back to work, just unwip it
-#
-# This function return a warning if the current branch is a wip
-function work_in_progress() {
-  if $(git log -n 1 2>/dev/null | grep -q -c "\-\-wip\-\-"); then
-    echo "WIP!!"
-  fi
-}
-# these alias commit and uncomit wip branches
-alias gwip='git add -A; git ls-files --deleted -z | xargs -r0 git rm; git commit -m "--wip--"'
-alias gunwip='git log -n 1 | grep -q -c "\-\-wip\-\-" && git reset HEAD~1'
+alias gm='git merge'
+alias gmom='git merge origin/master'
+alias gmt='git mergetool --no-prompt'
+alias gmtvim='git mergetool --no-prompt --tool=vimdiff'
+alias gmum='git merge upstream/master'
+
+alias gp='git push'
+alias gpd='git push --dry-run'
+alias gpf='git push --force'
+alias gpoat='git push origin --all && git push origin --tags'
+compdef _git gpoat=git-push
+alias gpu='git push upstream'
+alias gpv='git push -v'
+
+alias gr='git remote'
+alias gr1='git rebase -i HEAD~'
+alias gr2='git rebase -i HEAD~2'
+alias gr3='git rebase -i HEAD~3'
+alias gr4='git rebase -i HEAD~4'
+alias gr5='git rebase -i HEAD~5'
+alias gr6='git rebase -i HEAD~6'
+alias gr7='git rebase -i HEAD~7'
+alias gr8='git rebase -i HEAD~8'
+alias gr9='git rebase -i HEAD~9'
+alias gr10='git rebase -i HEAD~10'
+alias grb='git rebase'
+alias grba='git rebase --abort'
+alias grbc='git rebase --continue'
+alias grbi='git rebase -i'
+alias grbm='git rebase master'
+alias grbs='git rebase --skip'
+alias grh='git reset HEAD'
+alias grhh='git reset HEAD --hard'
+alias grmv='git remote rename'
+alias grrm='git remote remove'
+alias grset='git remote set-url'
+alias grt='cd $(git rev-parse --show-toplevel || echo ".")'
+alias gru='git reset --'
+alias grup='git remote update'
+alias grv='git remote -v'
+
+alias gsb='git status -sb'
+alias gsd='git svn dcommit'
+alias gsi='git submodule init'
+alias gsps='git show --pretty=short --show-signature'
+alias gsr='git svn rebase'
+alias gss='git status -s'
+alias gst='git status'
+alias gsta='git stash'
+alias gstaa='git stash apply'
+alias gstd='git stash drop'
+alias gstl='git stash list'
+alias gstp='git stash pop'
```

Comment:
I am using this function for this currently:

    gstp () {
      local log stash_ref
      if [[ "$1" = <-> ]] || [[ -z "$1" ]]
      then
        stash_ref="stash@{${1:-0}}" 
      else
        stash_ref="$1" 
      fi
      log=$(git log -g --pretty="%s" $stash_ref -1)  || {
        echo "Failed to display log for $stash_ref."
        return 1
      }
      local current_branch="$(current_branch)"
      if ! [[ $log = "WIP on "${current_branch}* ]]
      then
        echo "Warning: stash appears to be for another branch."
        echo "Current: ${current_branch}"
        echo "Log: $log"
        echo -n "Continue? "
        read -q || {
          echo
          return
        }
        echo
      fi
      git stash pop $stash_ref
    }